plenum/test/batching_3pc/catch-up/test_freeing_forwarded_not_preprepared_request.py

```diff
@@ -40,18 +39,15 @@ def test_freeing_forwarded_not_preprepared_request(
 
     sdk_send_batches_of_random_and_check(looper, txnPoolNodeSet, sdk_pool_handle,
                                          sdk_wallet_steward, CHK_FREQ, CHK_FREQ)
-    with delay_rules(behind_node.nodeIbStasher, chk_delay(delay=sys.maxsize,
-                                                          instId=behind_node.replicas.values()[-1])):
-        with delay_rules(behind_node.nodeIbStasher,
-                         ppDelay(delay=sys.maxsize),
-                         pDelay(delay=sys.maxsize),
-                         cDelay(delay=sys.maxsize)):
-            count = behind_node.spylog.count(behind_node.allLedgersCaughtUp)
+    count = behind_node.spylog.count(behind_node.allLedgersCaughtUp)
+    with delay_rules(behind_node.nodeIbStasher,
+                     ppDelay(delay=sys.maxsize),
+                     pDelay(delay=sys.maxsize),
+                     cDelay(delay=sys.maxsize)):
+        with delay_rules(behind_node.nodeIbStasher, chk_delay(delay=sys.maxsize)):
             sdk_send_batches_of_random(looper, txnPoolNodeSet, sdk_pool_handle,
                                        sdk_wallet_steward, req_num, req_num)
-            looper.run(eventually(node_caughtup, behind_node, count, retryWait=1))
             looper.run(eventually(lambda: assertExp(len(behind_node.requests) == req_num)))
-
-    # We execute caughtup requests
-    looper.run(eventually(lambda: assertExp(len(behind_node.requests) == req_num)))
-    assert all(r.executed for r in behind_node.requests.values() if behind_node.seqNoDB.get(r.request.key)[1])
+        # Start catchup with the quorum of Checkpoints
+        looper.run(eventually(node_caughtup, behind_node, count, retryWait=1))
+        looper.run(eventually(lambda: assertExp(len(behind_node.requests) == 0)))
```

Comment:
Why the test is changed so significantly?

Reply:
My test was incorrect because `behind_node.replicas.values()[-1]` is not instId and checkpoints are not delayed. It means that the catchup started early.